index.js

```diff
@@ -17,29 +17,30 @@ module.exports = function kindOf(val) {
   if (val === true || val === false || val instanceof Boolean) {
     return 'boolean';
   }
+
   if (typeof val !== 'object') {
+    // here: function, symbol, string (literal), number (literal)
     return typeof val;
   }
   if (Array.isArray(val)) {
     return 'array';
   }
-
-  var type = toString.call(val);
-
-  if (val instanceof RegExp || type === '[object RegExp]') {
+  if (val instanceof RegExp) {
     return 'regexp';
   }
-  if (val instanceof Date || type === '[object Date]') {
+  if (val instanceof Date) {
```

Comment:
> I dont see problem with instanceof for Date. Some cases from somewhere?

Like I mentioned, `instanceof` fails when the object originates from a different javascript context. It is possible for the object to be from a different instance of `Date`. to quote http://bonsaiden.github.io/JavaScript-Garden/

> One important thing to note here is that instanceof does not work on objects that originate from different JavaScript contexts (e.g. different documents in a web browser), since their constructors will not be the exact same object.